### PR TITLE
2.2 instance types

### PR DESCRIPTION
**Pull request by jameinel**
## Description of change

This is just a result of downloading the latest SKU list from Amazon (120MB) and then running 'go generate' to generate the new list of instance types.

Given the header, it appears we are actually 9 months out of date with the official SKU, so we need to update our general process to refresh more often than we're currently doing.

## QA steps

```
$ juju bootstrap aws/ap-northeast-1
$ juju deploy ubuntu --constraint instance-type=r4.4xlarge
```

Without this change, I *think* we won't know about r4.4xlarge. But I'm mostly working that out from the diff, not because I know from some other rationale.

## Documentation changes

We could make a note of supporting more types.

## Bug reference

[lp:1717181](https://bugs.launchpad.net/juju/+bug/1717181)

### CODE REVIEW

**Comment by anastasiamac** [comment]
Created a bug for this: https://bugs.launchpad.net/juju/+bug/1717181

File: provider/ec2/internal/ec2instancetypes/generated.go [hunk 1]
```diff
@@ -16,8 +16,8 @@ var (
 	both        = []string{arch.AMD64, arch.I386}
 )
 
-// Version: 20161213014831
-// Publication date: 2016-12-13 01:48:31 +0000 UTC
+// Version: 20170901182201
+// Publication date: 2017-09-01 18:22:01 +0000 UTC
```
Comment (anastasiamac) [inline provider/ec2/internal/ec2instancetypes/generated.go]: Wow - 9 months :D

**Comment by anastasiamac** [comment]
$$merge$$

**Comment by jameinel** [comment]
$$merge$$ updated the test that failed because we support more instance types now.

**Comment by nskaggs** [comment]
@jameinel this sounds like something we can put a ci test in for to check our types against the latest amazon sku. Is that doable, make sense? We could run it say once a day or week and alert when out of sync.

**Pull request merged**